vowpalwabbit/config/options_types.h

```diff
@@ -1,42 +0,0 @@
-// Copyright (c) by respective owners including Yahoo!, Microsoft, and
-// individual contributors. All rights reserved. Released under a BSD (revised)
-// license as described in the file LICENSE.
-
-#pragma once
-
-#include <vector>
-#include <string>
-#include <cstdint>
-#include <cstddef>
-
-namespace VW
-{
-namespace config
-{
-template <typename... Types>
-struct typelist
-{
-private:
-  template <class Head, class... Tail>
-  struct head_impl
-  {
-    using type = Head;
-  };
-  template <class Head, class... Tail>
-  struct tail_impl
-  {
-    using type = typelist<Tail...>;
-  };
-
-public:
-  using head = typename head_impl<Types...>::type;
-  using tail = typename tail_impl<Types...>::type;
-};
-
-// is_scalar detection relies on only accepting std::vector<T> for vector types. If that changes, update
-// is_scalar_impl below.
-using supported_options_types =
```

Comment:
is there any place where this list is explicit, or what's the status quo?

Reply:
Yes, this is now the canonical location https://github.com/VowpalWabbit/vowpal_wabbit/blob/598ef97820b12771952bd5f247b4be7671c9a28b/vowpalwabbit/config/option.h#L63